drivers/display/ssd1673.c

```diff
@@ -0,0 +1,512 @@
+		}
+
+	} else {
+		driver->numof_part_cycles++;
+		if (driver->last_lut != SSD1673_LAST_LUT_DEFAULT) {
+			driver->last_lut = SSD1673_LAST_LUT_DEFAULT;
+			if (ssd1673_write_cmd(driver, SSD1673_CMD_UPDATE_LUT,
+					      ssd1673_lut_default,
+					      sizeof(ssd1673_lut_default))) {
+				return -1;
+			}
+		}
+	}
+
+	tmp = (SSD1673_CTRL2_ENABLE_CLK |
+	       SSD1673_CTRL2_ENABLE_ANALOG |
+	       SSD1673_CTRL2_TO_PATTERN |
+	       SSD1673_CTRL2_DISABLE_ANALOG |
+	       SSD1673_CTRL2_DISABLE_CLK);
+	if (ssd1673_write_cmd(driver, SSD1673_CMD_UPDATE_CTRL2,
+			      &tmp, sizeof(tmp))) {
+		return -1;
+	}
+
+	if (ssd1673_write_cmd(driver, SSD1673_CMD_MASTER_ACTIVATION,
+			      NULL, 0)) {
+		return -1;
+	}
+
+	return 0;
+}
+
+int ssd1673_write_ram(const struct device *dev, const u16_t x, const u16_t y,
+		      const u16_t w, const u16_t h,
+		      const u16_t p, const void *data)
+{
+	size_t len = w * h / p;
+	struct ssd1673_data *driver = dev->driver_data;
+	u8_t cmd = SSD1673_CMD_WRITE_RAM;
+	u8_t dummy_page[SSD1673_RAM_YRES];
+	struct spi_buf buf = {.buf = &cmd, .len = 1};
+	struct spi_buf_set buf_set = {.buffers = &buf, .count = 1};
+	bool update = true;
+
+	if (!data || !len) {
+		return -1;
+	}
+
+	ssd1673_busy_wait(driver);
+	memset(dummy_page, 0xff, sizeof(dummy_page));
+
+	switch (driver->scan_mode) {
+	case SSD1673_DATA_ENTRY_XIYDY:
+		if (ssd1673_set_ram_param(driver,
+					  SSD1673_PANEL_FIRST_PAGE,
+					  SSD1673_PANEL_LAST_PAGE + 1,
+					  SSD1673_PANEL_LAST_GATE,
+					  SSD1673_PANEL_FIRST_GATE)) {
+			return -1;
+		}
+
+		if (ssd1673_set_ram_ptr(driver,
+					SSD1673_PANEL_FIRST_PAGE,
+					SSD1673_PANEL_LAST_GATE)) {
+			return -1;
+		}
+
+		break;
+
+	case SSD1673_DATA_ENTRY_XDYIY:
+		if (ssd1673_set_ram_param(driver,
+					  SSD1673_PANEL_LAST_PAGE + 1,
+					  SSD1673_PANEL_FIRST_PAGE,
+					  SSD1673_PANEL_FIRST_GATE,
+					  SSD1673_PANEL_LAST_GATE)) {
+			return -1;
+		}
+
+		if (ssd1673_set_ram_ptr(driver,
+					SSD1673_PANEL_LAST_PAGE + 1,
+					SSD1673_PANEL_FIRST_GATE)) {
+			return -1;
+		}
+
+		break;
+	default:
+		return -1;
+	}
+
+	gpio_pin_write(driver->dc, CONFIG_SSD1673_DC_PIN, 0);
+	if (spi_write(driver->spi_dev, &driver->spi_config, &buf_set)) {
+		return -1;
+	}
+
+	gpio_pin_write(driver->dc, CONFIG_SSD1673_DC_PIN, 1);
+	/* clear unusable page */
+	if (driver->scan_mode == SSD1673_DATA_ENTRY_XDYIY) {
+		buf.buf = dummy_page;
+		buf.len = sizeof(dummy_page);
+		if (spi_write(driver->spi_dev, &driver->spi_config, &buf_set)) {
+			return -1;
+		}
+	}
+
+	buf.buf = (u8_t *)data;
+	buf.len = len;
+	if (spi_write(driver->spi_dev, &driver->spi_config, &buf_set)) {
+		return -1;
+	}
+
+	/* clear unusable page */
+	if (driver->scan_mode == SSD1673_DATA_ENTRY_XIYDY) {
+		buf.buf = dummy_page;
+		buf.len = sizeof(dummy_page);
+		if (spi_write(driver->spi_dev, &driver->spi_config, &buf_set)) {
+			return -1;
+		}
+	}
+
+
+	if (update) {
+		if (driver->contrast) {
+			return ssd1673_update_display(dev, true);
```

Comment:
Is this setting any contrast or just triggering a part of the init sequence?

Reply:
It triggers initial update of EPD